policy-controller/k8s/index/src/index.rs

```diff
@@ -1070,13 +1079,53 @@ impl Pod {
             );
         };
 
+        let mut http_routes = HashMap::default();
+
+        // No Server is configured, so requests on the Pod's probe paths are
+        // authorized.
+        if let Some(ref probe_networks) = config.probe_networks {
+            http_routes = Self::http_probe_routes(probes, port, probe_networks.clone());
+        }
+
         InboundServer {
             reference: ServerRef::Default(policy.to_string()),
             protocol,
             authorizations,
-            http_routes: HashMap::default(),
+            http_routes,
         }
     }
+
+    fn http_probe_routes(
+        probes: &PortMap<HashSet<String>>,
+        port: NonZeroU16,
+        networks: Vec<IpNet>,
+    ) -> HashMap<String, InboundHttpRoute> {
+        let mut routes = HashMap::default();
+        for path in probes.get(&port).into_iter().flatten() {
+            let matches = vec![HttpRouteMatch {
+                path: Some(PathMatch::Exact(path.to_string())),
+                method: Some(Method::GET),
+                ..Default::default()
+            }];
+            let rules = vec![InboundHttpRouteRule {
+                matches,
+                ..Default::default()
+            }];
+            let client_authz = ClientAuthorization {
+                networks: networks.clone().into_iter().map(Into::into).collect(),
+                authentication: ClientAuthentication::Unauthenticated,
+            };
+            let mut authorizations = HashMap::default();
+            authorizations.insert(AuthorizationRef::Default("probe".to_string()), client_authz);
+            let route = InboundHttpRoute {
+                rules,
+                authorizations,
+                ..Default::default()
+            };
```

Comment:
tiny, unimportant style nit, take it or leave it: this crate already has a dev-dependency on the `maplit` crate: https://github.com/linkerd/linkerd2/blob/ae974aa1ea8590eb9a8dea2681d75b6301707d61/policy-controller/k8s/index/Cargo.toml#L22
so we could change that to a "real" dependency and use its macro here?
```suggestion
            let route = InboundHttpRoute {
                rules,
                authorizations: hashmap![
                    AuthorizationRef::Default("probe".to_string()) => client_authz,
                ],
                ..Default::default()
            };
```
(of course, we would need to change the dev-dep into a regular dependency in `Cargo.toml` as well)

Reply:
This won't work. This is an `AHashMap`